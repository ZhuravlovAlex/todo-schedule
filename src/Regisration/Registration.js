import React from "react";
import styles from "./Registration.module.css"

function Registration () {
	return (
		<form className={styles.registr} >
			<input 
				required
				placeholder='Please, enter your name'/>
			<button type="submit">Enter</button>
		</form>
	)
}

export default Registration