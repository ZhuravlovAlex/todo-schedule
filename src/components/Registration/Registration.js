import React from "react";
import styles from "./Registration.module.css"
import { signUp } from '../../store/action/signUp';

function Registration({ register, dispatch }) {

    let input;
    const submitHandler = e => {
        e.preventDefault();
        const text = input.value.trim();
        if (!text) return

        dispatch(signUp());
    }

    return register ? <div></div> :
        <form className={styles.registr} onSubmit={submitHandler}>
            <input
                required
                placeholder='Please, enter your name'
                ref={node => input = node} />
            <button type="submit">Enter</button>
        </form>
}

export default Registration;