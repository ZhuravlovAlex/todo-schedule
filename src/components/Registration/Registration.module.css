.registr {
	/* display: flex;
	justify-content: space-between;
	align-items: center;
	padding: 0.5rem 1 rem;
	border: 1px solid #ccc;
	border-radius: 4px; */
	margin-bottom: 1rem
}