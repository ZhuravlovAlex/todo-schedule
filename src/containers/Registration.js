import { connect } from "react-redux";
import Registration from "../components/Registration/Registration";

const mapStateToProps = state => ({
	register: state.register
})

export default connect(
	mapStateToProps, null
)(Registration)