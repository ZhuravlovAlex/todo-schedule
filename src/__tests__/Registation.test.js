import renderer from 'react-test-renderer';
import Registration from '../components/Registration/Registration';


it('renders correctly when not signed up ', () => {
	const tree = renderer.create(<Registration
		register={false}
		dispatch={() => {}}
	/>).toJSON();

	expect(tree).toMatchSnapshot();
});

it('renders correctly when signed up ', () => {
	const tree = renderer.create(<Registration
		register={true}
		dispatch={() => {}}
	/>).toJSON();

	expect(tree).toMatchSnapshot();
});


