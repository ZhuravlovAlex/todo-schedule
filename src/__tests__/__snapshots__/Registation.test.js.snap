// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not signed up  1`] = `
<form
  className="registr"
  onSubmit={[Function]}
>
  <input
    placeholder="Please, enter your name"
    required={true}
  />
  <button
    type="submit"
  >
    Enter
  </button>
</form>
`;

exports[`renders correctly when signed up  1`] = `<div />`;
